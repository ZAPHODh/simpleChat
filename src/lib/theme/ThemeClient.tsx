'use client'

import { ThemeProvider } from 'styled-components'
import { theme } from './theme'

type ThemeClientProps = {
    children: React.ReactNode
}

export const ThemeClient = ({ children }: ThemeClientProps) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}
