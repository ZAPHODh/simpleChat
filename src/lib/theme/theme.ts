import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
    colors: {
        white: '#ffffff ',
        darker: '#0B0B0D',
        dark: '#181619',
        normalDark: '#181B22',
        brightDark: '#272A31',
        gold: '#A76B09',
        brightGold: '#DDA74F',
    },
    imgBorderEffects: {
        borderThickness: '10px',
        offset: '20px',
        gap: '5px',
    },
}
