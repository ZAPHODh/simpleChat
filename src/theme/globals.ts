'use client'

import { Montserrat } from 'next/font/google'
import { createGlobalStyle } from 'styled-components'
import { css } from 'styled-components'

const montserrat = Montserrat({
    subsets: ['latin'],
    fallback: ['Georgia', 'ui-serif', 'serif'],
    weight: ['300', '700'],
})

export const GlobalStyles = createGlobalStyle`${() => css`
    html,
    body {
        width: 100%;
        max-width: 100vw;
        padding: 0;
        margin: 0;
        font-family: ${montserrat.style.fontFamily};
        overflow-x: hidden;
    }

    main {
        width: 100%;
        max-width: 100vw;
        padding: 0;
        margin: 0;
    }

    input {
        font-family: inherit;
    }

    a {
        color: inherit;
        text-decoration: none;
    }

    * {
        box-sizing: border-box;
    }

    button {
        font-family: inherit;
    }
`}`
