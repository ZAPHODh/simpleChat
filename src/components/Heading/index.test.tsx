import { render } from '@testing-library/react'
import { Heading } from '.'
import { ThemeClient } from '../../theme/ThemeClient'

describe('Heading', () => {
    it('should renders a heading', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Heading>test</Heading>
            </ThemeClient>
        )

        const heading = getByRole('heading')

        expect(heading).toBeInTheDocument()
    })

    it('should renders heading with the correct "as" propperty', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Heading as="h2">test</Heading>
            </ThemeClient>
        )

        const heading = getByRole('heading')

        expect(heading.tagName).toBe('H2')
    })
    it('should renders heading with the correct children text', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Heading>children text</Heading>
            </ThemeClient>
        )

        const heading = getByRole('heading')

        expect(heading).toHaveTextContent('children text')
    })

    it('should match snapshot', () => {
        const { getByRole } = render(
            <ThemeClient>
                <Heading>test</Heading>
            </ThemeClient>
        )

        const heading = getByRole('heading')

        expect(heading).toMatchSnapshot()
    })
})
