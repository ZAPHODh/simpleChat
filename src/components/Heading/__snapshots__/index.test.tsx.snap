// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshot 1`] = `
<h1
  class="sc-ba81c2ff-0 jFApBo"
>
  test
</h1>
`;
